blockchain/chain_sync/src/validation.rs

```diff
@@ -75,65 +75,73 @@ impl<'a> TipsetValidator<'a> {
         for block in self.0.blocks() {
             self.validate_msg_root(&chainstore.db, block)?;
             if let Some(bad) = bad_block_cache.peek(block.cid()).await {
-                return Err(TipsetValidationError::InvalidBlock(*block.cid(), bad));
+                return Err(Box::new(TipsetValidationError::InvalidBlock(
+                    *block.cid(),
+                    bad,
+                )));
             }
         }
 
         Ok(())
     }
 
-    #[allow(clippy::result_large_err)]
     pub fn validate_epoch(
         &self,
         genesis_tipset: Arc<Tipset>,
         block_delay: u64,
-    ) -> Result<(), TipsetValidationError> {
+    ) -> Result<(), Box<TipsetValidationError>> {
         let now = SystemTime::now()
             .duration_since(UNIX_EPOCH)
             .unwrap()
             .as_secs();
         let max_epoch = ((now - genesis_tipset.min_timestamp()) / block_delay) + MAX_HEIGHT_DRIFT;
         let too_far_ahead_in_time = self.0.epoch() as u64 > max_epoch;
         if too_far_ahead_in_time {
-            Err(TipsetValidationError::EpochTooLarge)
+            Err(Box::new(TipsetValidationError::EpochTooLarge))
         } else {
             Ok(())
         }
     }
 
-    #[allow(clippy::result_large_err)]
     pub fn validate_msg_root<DB: BlockStore>(
         &self,
         blockstore: &DB,
         block: &Block,
-    ) -> Result<(), TipsetValidationError> {
+    ) -> Result<(), Box<TipsetValidationError>> {
         let msg_root = Self::compute_msg_root(blockstore, block.bls_msgs(), block.secp_msgs())?;
         if block.header().messages() != &msg_root {
-            Err(TipsetValidationError::InvalidRoots)
+            Err(Box::new(TipsetValidationError::InvalidRoots))
         } else {
             Ok(())
         }
     }
 
-    #[allow(clippy::result_large_err)]
     pub fn compute_msg_root<DB: BlockStore>(
         blockstore: &DB,
         bls_msgs: &[Message],
         secp_msgs: &[SignedMessage],
-    ) -> Result<Cid, TipsetValidationError> {
+    ) -> Result<Cid, Box<TipsetValidationError>> {
         // Generate message CIDs
         let bls_cids = bls_msgs
             .iter()
             .map(Cbor::cid)
-            .collect::<Result<Vec<Cid>, fvm_ipld_encoding::Error>>()?;
+            .collect::<Result<Vec<Cid>, fvm_ipld_encoding::Error>>()
+            .map_err(Into::into)
+            .map_err(Box::new)?;
```

Comment:
Can't you add something like:
```rust
impl From<EncodingError> for Box<TipsetValidationError> {
    fn from(err: EncodingError) -> Self {
        Box::new(TipsetValidationError::Encoding(err))
    }
}
```

And then the conversion happens automatically?

Reply:
Done